[16.0][IMP] account-financial-tools: ADD account_loan_analytic_account

lmarion-source opened the pull request:
This module allows to add analytic distributions on loans and propagate them  to moves when the loan is posted

acsonefho commented on account_loan_analytic_account/models/account_move_line.py:
```diff
@@ -0,0 +1,36 @@
+                and loan
+                and vals["account_id"] == loan.interest_expenses_account_id.id
+            ):
+                vals["analytic_distribution"] = loan.analytic_distribution
```
And what if the user already define some `analytic_distribution`? You completely overwrite filled values.

lmarion-source replied:
The analytic_distribution on the move_line must follow the one on the loan : if the distribution changes on the loan, the distribution should be erased on the move_line

acsonefho commented on account_loan_analytic_account/models/account_move_line.py:
```diff
@@ -0,0 +1,36 @@
+                and loan
+                and vals["account_id"] == loan.interest_expenses_account_id.id
+            ):
+                vals["analytic_distribution"] = loan.analytic_distribution
```
Same here, during the `write`.

lmarion-source replied:
same here than for the create, the user should not be able to change the distribution on the move_line independently of the loan

AnizR reviewed (approved):
Code LGTM

lmarion-source committed:
[16.0][IMP] account-financial-tools: ADD account_loan_analytic_account

    This module allows to add analytic distributions on loans and propagate them
    to moves when the loan is posted

sbidoul commented on account_loan_analytic_account/models/account_move_line.py:
```diff
@@ -0,0 +1,23 @@
+
+    loan_id = fields.Many2one(
+        related="move_id.loan_id",
+        store=True,
```
Why store=True? And actually, why adding a field here?

At first sight this PR does not require it

AnizR reviewed (approved):
Code looks good to me, thanks for the improvement 🚀
